src/mapping/toroidal_to_cylindrical.hpp

```diff
@@ -0,0 +1,442 @@
+#include <iostream>
+
+#include <ddc/ddc.hpp>
+
+#include "ddc_alias_inline_functions.hpp"
+#include "ddc_aliases.hpp"
+#include "mapping_tools.hpp"
+#include "math_tools.hpp"
+#include "tensor.hpp"
+#include "view.hpp"
+
+/**
+ * @brief A class for describing 3D mappings from the logical domain (toroidal one @f$(\rho, \theta, \varphi)@f$) 
+ * to the physical domain (cylindrical domain @f$(R, \varphi, Z)@f$ where 
+ *  - @f$ R @f$ represents the distance from the major axis (the central axis of the torus) to a point in the tokamak,
+ *  - @f$ \varphi @f$ is the toroidal angle and
+ *  - @f$ Z @f$ represents the vertical coordinate (i.e the height above or below the midplane of the tokamak.
+ *
+ * To keep on handling a direct set of coordinates (positive signature), it is convenient to reorder the 
+ * cylindrical coordinates as @f$(R, Z, \zeta)@f$, which is direct provided that @f$\zeta@f$ is chosen anti-trigonometric. 
+ * The subset @f$(R, Z)@f$ behaves as a set of two Cartesian coordinates. A 2D problem is then isolated by
+ * determining the metric tensor associated with the change of variables @f$(R, Z)\rightarrow(\rho,\theta)@f$.
+ *
+ * The elements of the reduced metric 2D tensor @f$\begin{bmatrix} g_{\rho\rho} & g_{\rho\theta} \\ g_{\theta\rho} & g_{\theta\theta} \end[bamtrix}@f$
+ * can be calculated by using the expressions of @f$ R(\rho,\theta) @f$ and @f$ Z(\rho,\theta)@f$ and calculating their partial derivatives. 
+ * The other coefficients satisfy @f$g_{\zeta\zeta} = R^2@f$ , and @f$g_{zeta\rho} = g_{\zeta\theta} = 0@f$.
+ *
+ * The 2D mapping describe here is only defined on a grid. Then this class decomposes the 2D mapping
+ * on B-splines to evaluate it on the physical domain.
+ *
+ * @f$ R(\rho,\theta) = \sum_k c_{R,k} B_k(\rho,\theta),@f$
+ *
+ * @f$ Z(\rho,\theta) = \sum_k c_{Z,k} B_k(\rho,\theta).@f$
+ *
+ */
+template <
+        class R,
+        class Z,
+        class Zeta class SplineEvaluator,
+        class Rho = typename SplineEvaluator::continuous_dimension_type1,
+        class Theta = typename SplineEvaluator::continuous_dimension_type2,
+        class MemorySpace = typename SplineEvaluator::memory_space>
+class ToroidalToCylindrical
+{
+    static_assert(std::is_same_v<MemorySpace, typename SplineEvaluator::memory_space>);
+
+public:
+    /**
+     * @brief Indicate the bspline type of the first logical dimension.
+     */
+    using BSplineRho = typename SplineEvaluator::bsplines_type1;
+    /**
+     * @brief Indicate the bspline type of the second logical dimension.
+     */
+    using BSplineTheta = typename SplineEvaluator::bsplines_type2;
+
+    /// @brief Indicate the first physical coordinate.
+    using cartesian_tag_R = R;
+    /// @brief Indicate the second physical coordinate.
+    using cartesian_tag_Z = Z;
+    /// @brief Indicate the first logical coordinate.
+    using curvilinear_tag_rho = Rho;
+    /// @brief Indicate the second logical coordinate.
+    using curvilinear_tag_theta = Theta;
```

Comment:
We need to have a think about these names. They are sometimes used in other files. E.g here:
https://github.com/gyselax/gyselalibxx/blob/95f8db61f1440a54589d1452061b4461bc3d0312/src/interpolation/polar_splines/polar_bsplines.hpp#L218-L219

so we may need a global naming convention for this or a better way to specify this in the files where it is used